include/swift/SIL/SILInstruction.h

```diff
@@ -2816,6 +2816,9 @@ class FunctionRefInst : public FunctionRefBaseInst {
                   TypeExpansionContext context);
 
 public:
+  /// Return the referenced function.
```

Comment:
I had intentionally removed `getReferencedFunction` and replaced it with `getInitiallyReferencedFunction` in `FunctionRefBaseInst` to make clear to clients that the referenced function returned might not actually what is dynamically called (e.g. in case of `DynamicFunctionRefInst`)

Why do you need to reintroduce it? The presence of `getReferencedFunction` makes the distinction murky.


```
  /// Return the referenced function if this is a function_ref instruction and  
  /// therefore a client can rely on the dynamically called function being equal
  /// to the returned value and null otherwise.                                 
  SILFunction *getReferencedFunctionOrNull() const {                            
    auto kind = getKind();                                                      
    if (kind == SILInstructionKind::FunctionRefInst)                            
      return f;                                                                 
    assert(kind == SILInstructionKind::DynamicFunctionRefInst ||                
           kind == SILInstructionKind::PreviousDynamicFunctionRefInst);         
    return nullptr;                                                             
  }                                                                             
                                                                                
  /// Return the initially referenced function.                                 
  ///                                                                           
  /// WARNING: This not necessarily the function that will be called at runtime.
  /// If the callee is a (prev_)dynamic_function_ref the actual function called 
  /// might be different because it could be dynamically replaced at runtime.   
  ///                                                                           
  /// If the client of this API wants to look at the content of the returned SIL
  /// function it should call getReferencedFunctionOrNull() instead.            
  SILFunction *getInitiallyReferencedFunction() const { return f; }   
```

Reply:
But this is not the case for the derived FunctionRefInst, right?
Not having it on the base class makes sense, but why not in FunctionRefInst?

Reply:
The reason why I re-introduced it was that both names, `getReferencedFunctionOrNull` and `getInitiallyReferencedFunction()` don't really make sense for the "static" `function_ref`

Reply:
I mis-parsed the line `class FunctionRefInst : public FunctionRefBaseInst {` even though I had read your commit message (which also says `FunctionRefInst`) I was under the impression this added it to the base.

Ignore me. This is of course fine.